geoips/filenames/base_paths.py

```diff
@@ -1,252 +1,171 @@
-elif getenv("GEOIPS_ANCILDAT"):
-    PATHS["GEOIPS_ANCILDAT"] = getenv("GEOIPS_ANCILDAT").rstrip("/")
-# Separating the auto-generated files from the source files allows for
-# individual users to read from the shared ancildat, and write to their own
-# auto-generated location
-if not getenv("GEOIPS_ANCILDAT_AUTOGEN"):
-    PATHS["GEOIPS_ANCILDAT_AUTOGEN"] = pathjoin(
-        PATHS["GEOIPS_OUTDIRS"], "ancildat_autogen"
+    # Output Directories
+    if not os.getenv("GEOIPS_OUTDIRS"):
+        raise KeyError(
+            "GEOIPS_OUTDIRS must be set in your environment. "
+            "Please set GEOIPS_OUTDIRS and try again"
+        )
+    paths["GEOIPS_OUTDIRS"] = os.getenv("GEOIPS_OUTDIRS").rstrip("/")
+    paths["GEOIPS_PACKAGES_DIR"] = os.path.abspath(
+        os.path.join(paths["BASE_PATH"], os.pardir, os.pardir)
     )
-elif getenv("GEOIPS_ANCILDAT_AUTOGEN"):
-    PATHS["GEOIPS_ANCILDAT_AUTOGEN"] = getenv("GEOIPS_ANCILDAT_AUTOGEN").rstrip("/")
-
-if not getenv("LOGDIR"):
-    PATHS["LOGDIR"] = pathjoin(PATHS["GEOIPS_OUTDIRS"], "logs")
-else:
-    PATHS["LOGDIR"] = getenv("LOGDIR").rstrip("/")
-
-if not getenv("GEOIPSDATA"):
-    PATHS["GEOIPSDATA"] = pathjoin(PATHS["GEOIPS_OUTDIRS"], "geoipsdata")
-else:
-    PATHS["GEOIPSDATA"] = getenv("GEOIPSDATA").rstrip("/")
-
-if not getenv("GEOIPS_VERS"):
-    PATHS["GEOIPS_VERS"] = "0.0.0"
-else:
-    PATHS["GEOIPS_VERS"] = getenv("GEOIPS_VERS")
-
-
-if getenv("TCWWW"):
-    PATHS["TCWWW"] = getenv("TCWWW").rstrip("/")
-else:
-    PATHS["TCWWW"] = pathjoin(PATHS["GEOIPS_OUTDIRS"], "preprocessed", "tcwww")
-
-if getenv("TCWWW_URL"):
-    PATHS["TCWWW_URL"] = getenv("TCWWW_URL").rstrip("/")
-else:
-    PATHS["TCWWW_URL"] = PATHS["TCWWW"]
-
-if getenv("PUBLICWWW"):
-    PATHS["PUBLICWWW"] = getenv("PUBLICWWW").rstrip("/")
-else:
-    PATHS["PUBLICWWW"] = pathjoin(PATHS["GEOIPS_OUTDIRS"], "preprocessed", "publicwww")
-
-if getenv("PUBLICWWW_URL"):
-    PATHS["PUBLICWWW_URL"] = getenv("PUBLICWWW_URL").rstrip("/")
-else:
-    PATHS["PUBLICWWW_URL"] = PATHS["PUBLICWWW"]
-
-if getenv("PRIVATEWWW"):
-    PATHS["PRIVATEWWW"] = getenv("PRIVATEWWW").rstrip("/")
-else:
-    PATHS["PRIVATEWWW"] = pathjoin(
-        PATHS["GEOIPS_OUTDIRS"], "preprocessed", "privatewww"
+    paths["GEOIPS_BASEDIR"] = get_env_var(
+        "GEOIPS_BASEDIR",
+        os.path.abspath(os.path.join(paths["GEOIPS_PACKAGES_DIR"], os.pardir)),
     )
 
-if getenv("PRIVATEWWW_URL"):
-    PATHS["PRIVATEWWW_URL"] = getenv("PRIVATEWWW_URL").rstrip("/")
-else:
-    PATHS["PRIVATEWWW_URL"] = PATHS["PRIVATEWWW"]
-
-PATHS["TC_DECKS_DB"] = pathjoin(
-    PATHS["GEOIPS_OUTDIRS"], "longterm_files", "tc", "tc_decks.db"
-)
-if getenv("TC_DECKS_DB"):
-    PATHS["TC_DECKS_DB"] = getenv("TC_DECKS_DB")
-
-if getenv("TC_DECKS_DIR"):
-    PATHS["TC_DECKS_DIR"] = getenv("TC_DECKS_DIR").rstrip("/")
-else:
-    PATHS["TC_DECKS_DIR"] = pathjoin(
-        PATHS["GEOIPS_OUTDIRS"], "longterm_files", "tc", "decks"
-    )
+    geoips_global_variables = {
```

Comment:
Maybe leave a comment here that these are the explicitly defined global variables (not dependent on another variable as a parent directory), pulling from os.getenv if it is specified, else use the defined default.

Reply:
Maybe these should also use "get_env_var" instead of specifying the defaults directly in the dict, for consistency and future-proofing for when we are also pulling from the YAML/JSON config?  If we udpated this, then these should probably be named "geoips_global_variables_defaults", because the actual paths would be set in a loop below calling "get_env_var".

Reply:
You're right. Using `get_env_var` makes way more sense here. My bad! Will fix.

Reply:
Fixed!

Reply:
It should probably be named "default_geoips_global_variables" for consistency?  Because the final values are set below?